### ISSUE TITLE
[Azure AKS] Autoscaler can't set agent pool size on an AKS cluster with AzureAD integration enabled

### ISSUE THREAD

**Issue by brantb** [kubernetes/autoscaler#1412]
Kubernetes version: 1.11.4 (Latest available on AKS as of writing)
cluster-autoscaler version: 1.3.4

After scaling down the cluster and deleting a VM, the Azure provider attempts to update the node count in the AKS agent pool. When the AKS cluster has the AzureAD integration enabled, this fails with the following errors:

```
I1116 20:38:15.436203       1 utils.go:422] Decreasing size of agentpool, expected=5 current=3 delta=-2
I1116 20:38:15.873178       1 azure_container_service_pool.go:304] Setting size for cluster ("development-001") with new count (3)
I1116 20:38:16.322277       1 azure_container_service_pool.go:184] Current size: 5, Target size requested: 3
E1116 20:38:16.322644       1 azure_container_service_pool.go:193] Failed to update AKS cluster (%q): %vdevelopment-001containerservice.ManagedClustersClient#CreateOrUpdate: Invalid input: autorest/validation: validation failed: parameter=parameters.ManagedClusterProperties.AadProfile.ServerAppSecret constraint=Null value=(*string)(nil) details: value can not be null; required parameter
E1116 20:38:16.322736       1 static_autoscaler.go:190] Failed to fix node group sizes: Failed to decrease agentpool: containerservice.ManagedClustersClient#CreateOrUpdate: Invalid input: autorest/validation: validation failed: parameter=parameters.ManagedClusterProperties.AadProfile.ServerAppSecret constraint=Null value=(*string)(nil) details: value can not be null; required parameter
```

I have a feeling this is happening because `AadProfile.ServerAppSecret` is a write-only property in the ARM API so it isn't being round-tripped when `managedContainerServicesClient.Get()` [is called earlier in the function](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/azure/azure_container_service_pool.go#L121-L123).

**Comment by brantb** [comment]
It looks like this is the same underlying issue as a few others (#1304 #1105 #1369) but I didn't search adequately. #1415 should fix this once it is released.

**Comment by feiskyer** [comment]
Fixed in the https://github.com/kubernetes/autoscaler/pull/1415, it should be included in next CA release (v1.3.5).
